load-gen/src/main/java/io/smartcat/data/loader/api/RateGenerator.java

```diff
@@ -0,0 +1,6 @@
+package io.smartcat.data.loader.api;
+
+public interface RateGenerator {
+
+    long getRate(long prevInstant, long currentInstant);
```

Comment:
You are right about the second variant of `getRate`. Not needed.
Also, it seems that I don't understand the idea behind RateGenerator.
My assumption is that it #getRate should return the rate value (e.g. 1000 msg/sec) for the given time instant. And that's all. In that case, the constant rate generator simply returns a constant number (e.g. 1000) for every `getRate` call. Simple as that. Basically it would ignore the passed time instant. In case of a more complex rate generator, the time instant would contribute to the current rate implementation.